### ISSUE TITLE
Callback invoked twice instead of once on a content when using IterateContents()

### ISSUE THREAD

**Issue by pkj415** [kopia/kopia#430]
When iterating contents as part of any process (let’s say GC), the intention is to iterate over a point in time view of the contents. However this happens as of now - iterate through all uncommitted contents and then iterate through all indices (representing committed contents). Invoking the callback on an uncommitted content might result in the content being committed (if the callback itself results in a flush) resulting in the callback being invoked later on the same content again when iterating over the committed set.

In a nutshell, a callback might get invoked twice on a content - once when it is in the uncommitted list and once when it is in the committed list.
